### DISCUSSION TITLE
Why not make courses a single large YAML file?

### DISCUSSION THREAD

**Discussion by grondilu** [kantord/LibreLingo#1204] [Ideas]
It's not fun to have to browse directories.

YAML already allows a hierarchical structure, adding one in the file system seems redundant.

A single file would be conceptually simpler and more convenient to edit IMHO

**Comment by kantord** [comment]
I would say

- File structures are harder to mess up than syntax so transferring some of the complexity to folder structures means reducing the possibility for syntax errors or other mistakes 
- Deeper hierarchical structures would require more complex indentation. That is not fun to deal with. And it's only worse when the file is large
- The file could be really big, it could be hard to work with with certain text editors/if someone is not experienced with working with large files
- It's easier to get lost in one large file
- If you want to edit a specific skill it might be much easier to actually see a manageable amount of data on your screen at a time
- One large file is harder to deal with when multiple people are editing it at the same time. It's easier to merge small scale changes in small files that potentially avoid having merge conflicts 

Maybe there are some potential alternative solutions to the problem of navigation difficulties?

I have some ideas:

- Having a GUI tool for editing skills
- Having a gui where you can do a full text search in course content
- Having some tool that can visualize the full folder structure on a single, scrollable screen
